File: video2-single-responsibility/python/macbook_v1.py
class LaptopProperty: 
    cpu = ""
    size = 0
    resolution = ""

class LaptopFunction:
    def startUp(self): pass
    def shutDown(self): pass

class MacLaptopProperty(LaptopProperty):
    cpu = ""
    size = 0
    resolution = ""
    
class MacLaptopFunction(LaptopFunction):
    def startUp(self): pass
    def shutDown(self): pass

class MacBookV1:
    property: MacLaptopProperty
    function: MacLaptopFunction
    
    def __init__(self, p: MacLaptopProperty, f: MacLaptopFunction):
        self.property = p
        self.function = f

MacBookV1(MacLaptopProperty(), MacLaptopFunction())

File: v2_single_responsibility/java/MacBookV1.java
public class MacBookV1 {

	private MacLaptopProperty property;
	private MacLaptopFunction function;
    
    public static void main(String[] args) {
        System.out.println("Hello");   
    }
	public MacBookV1(MacLaptopProperty p, MacLaptopFunction f) {
		this.property = p;
		this.function = f;
	}
}


interface LaptopProperty {
	String cpu = "";
	int size = 0;
	String resolution = "";
}
interface LaptopFunction {
	void startUp();
	void shutDown();
}

class MacLaptopProperty implements LaptopProperty {
	String cpu = "";
	int size = 0;
	String resolution = "";
}

class MacLaptopFunction implements LaptopFunction {
	public void startUp() {};
	public void shutDown() {};
}
